packages/core/src/data_sources/model/DataResolverListener.ts

```diff
@@ -100,6 +104,14 @@ export default class DataResolverListener {
       this.createListener(em, `${DataSourcesEvents.path}:${normPath}`),
     );
 
+    dataListeners.push(
+      this.createListener(em, DataSourcesEvents.path, ({ path: eventPath }: { path: string }) => {
+        if (eventPath.startsWith(path)) {
+          this.onChange();
+        }
+      }),
+    );
```

Comment:
Do you think we really need this? 🤔

I guess I understand the reason behind it but it seems it could trigger too many unnecessary checks. When we add data components, I'd expect them to be already listening to the exact paths.